### DISCUSSION TITLE
Bypassing GitHub Ruleset rules for automerge

### DISCUSSION THREAD

**Discussion by jamesbraza** [renovatebot/renovate#31620] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I am using Renovate and have enabled `automerge` capability. I know it works because the PR was opened with auto-merge enabled:

![image](https://github.com/user-attachments/assets/a8006778-2954-4ed9-8f35-09734ae856c0)

However, I am getting blocked by PR approval count thresholds from our GitHub Ruleset.

It looks like the docs added in https://github.com/renovatebot/renovate/pull/16370 are no longer relevant with GitHub Rulesets. I have tried adding Renovate to the "Bypass list", but it does not seem to work:

![image](https://github.com/user-attachments/assets/68236659-8b11-4612-842e-b4cf77f897e0)

How do people allow Renovate to bypass GitHub Rulesets?

Alternately, is the only way to handle this to use https://github.com/apps/renovate-approve? If possible I would like to avoid adding another App to my ecosystem.

---

https://github.com/renovatebot/renovate/discussions/23612 exists that is requesting information on this.

**Comment by rarkins** [comment, marked as answer]
The renovate approve apps exist exactly for this purpose. They need to be separate apps because github requires two _different_ identities to approve. It can't be the same app approving itself twice.

It's also possible that github might have some way you can get around this but it's hard to keep up with their rules for this.

**Reply by jamesbraza** [reply]
Perfect thanks for confirming 👍
